web/enhanced_logging.py

```diff
@@ -0,0 +1,57 @@
+import random
+import logging
+from sanic import request
+
+logger = logging.getLogger(__name__)
+
+
+def get_random_hash():
+    hash = random.getrandbits(64)
+    return "{0:016x}".format(hash)
+
+def log_d(request, message):
+    try:
+        func_id = request.cookies["__log_extension"]
+        logger.debug("{} {}".format(func_id, message))
+    except:
+        logger.debug(message)
+
+def log_w(request, message):
+    try:
+        func_id = request.cookies["__log_extension"]
+        logger.warn("{} {}".format(func_id, message))
+    except:
+        logger.warn(message)
+
+def log_i(request, message):
+    try:
+        func_id = request.cookies["__log_extension"]
+        logger.info("{} {}".format(func_id, message))
+    except:
+        logger.info(message)
+
+def log_e(request, message):
+    try:
+        func_id = request.cookies["__log_extension"]
+        logger.error("{} {}".format(func_id, message))
+    except:
+        logger.error(message)
+
+
+def log_func_boundaries(func):

```

Comment:
I think should be renamed, any idea Herr Pisar?

Reply:
I think the name is actually very good!